### DISCUSSION TITLE
help configure bmp180

### DISCUSSION THREAD

**Discussion by marsaiz** [cyberman54/ESP32-Paxcounter#1018] [General]
Hello, I cannot configure a bmp180 sensor included in the project. I think I have problems with a conflict in the i2c port with the OLED screen of my Lilygo LORA32 T3_ v1.6.1. Could I change the SDA and SCL pin to connect the BMP180 sensor to other GPIOs? thank you so much. A big hug.

**Comment by cyberman54** [comment]
There cannot be a conflict on the i2c bus, as long as BMP180 and display use different addresses. Check zur wiring of BMP180 which selects it's address, then put this address in the hal file.

**Reply by marsaiz** [reply]
Hola, el cableado esta correcto en los pines 21(SDA) y 22 (SCL). Las direcciones son 0x3C la pantalla y 0x77 el sensor. la Libreria Adrafuir_BMP085.h tiene la dirección correcta 
#define BMP085_I2CADDR 0x77 //!< BMP085 I2C address
En el archivo ttgov21new.h tengo agreagadas las lineas: // BMP180 sensor on I2C bus
//define HAS_BME 1 // Enable BME sensors in general
//#define HAS_BMP180
#define BMP180_ADDR 0x77
Pero todavía no envía por el puerto 7 los datos del sensor, le paso los datos del monitor:
/src/Wire.cpp: Bus already started in Master Mode.
08:53:56.184 > I (269) src/i2c.cpp: Device found at 0x3C, type = Unknown
08:53:56.198 > I (278) src/i2c.cpp: Device found at 0x77, type = BMP180
08:53:56.199 > I (278) src/i2c.cpp: 2 I2C device(s) found
08:53:56.199 > W (279) /home/marcelo/.platformio/packages/framework-arduinoespressif32/libraries/Wire/src/Wire.cpp: Bus already started in Master Mode.
08:54:06.346 > I (10426) src/main.cpp: Starting LED Controller...
08:54:06.347 > I (10427) src/power.cpp: ADC characterization based on reference voltage stored in eFuse
08:54:06.358 > I (10430) src/main.cpp: Starting libpax...
08:54:06.358 > I (10431) src/main.cpp: WIFISCAN: on
08:54:06.358 > I (10434) src/main.cpp: BLESCAN: off
08:54:06.452 > I (10532) src/main.cpp: Starting rcommand interpreter...
08:54:06.452 > I (10532) src/rcommand.cpp: Rcommand send queue created, size 55 Bytes
08:54:06.459 > I (10532) src/lorawan.cpp: LORA send queue created, size 530 Bytes
08:54:06.474 > I (10554) src/lorawan.cpp: Starting LMIC...
08:54:06.475 > I (10556) src/main.cpp: Starting Interrupt Handler...
08:54:06.475 > I (10556) src/main.cpp: Starting Timers...
08:54:06.508 > I (10587) src/timekeeper.cpp: Starting time pulse...
08:54:06.508 > I (10588) src/timekeeper.cpp: Timepulse: internal (ESP32 hardware timer)
08:54:07.503 > I (11588) src/main.cpp: Features: DISP LED BATT LORA PLAIN TIME
08:54:07.534 > I (11616) src/timesync.cpp: [11.622] Timeserver sync request started, seqNo#19
08:54:25.618 > I (29697) src/lorawan.cpp: DEVaddr: 0x260C06A2 | Network ID: 0x000013 | Network Type: 0
08:54:25.618 > I (29698) src/lorawan.cpp: RSSI: -31 | SNR: 0
08:54:25.625 > I (29698) src/lorawan.cpp: Radio parameters: SF10 | BW125 | CR 4/5
08:54:34.020 > I (38103) src/timekeeper.cpp: [38.108] UTC time: 1721044474.000 sec
08:54:40.240 > I (44325) src/lorawan.cpp: 1 byte(s) sent to LORA

Necesito realizar alguna otra modificación?  Muchas gracias...

**Reply by cyberman54** [reply]
Enable BME data transmission by sending rcommand 0x85 0x02 0x21 on Port 2.

**Reply by marsaiz** [reply]
hola, lo siento el monitor muestra este erro: 
10:22:59.079 > I (495867) src/rcommand.cpp: Remote command: get BME sensor data
10:22:59.085 > W (495867) src/rcommand.cpp: BME sensor not supported.

Podría indicarme algun paso a seguir? muchas gracias

**Reply by cyberman54** [reply]
Then you did not proper configure BMP180 with your build. Check you hal file. Rebuild the code. Clear the flash before reflashing.

**Reply by marsaiz** [reply]
Muchas gracias por sus respuestas, voy a seguir consultando (espero que no sea mucha molestia). Usted se refiere al archivo hal de la placa Lilygo LORA32 que estoy aplicando ttgov21new.h ? y con respecto al código paxcounter.conf solamente aplica a el sensor BME680, por que no mensiona otro modelo de sensor.
